src/anyon/run_jobs.jl

```diff
@@ -0,0 +1,549 @@
+    generation    ::String
+    serial_number ::String
+    printout_delay::Real    =200. # milliseconds between get_status() printouts
+end
+
+get_client(qpu_service::AnyonQPU)=qpu_service.client
+get_printout_delay(qpu_service::AnyonQPU)=qpu_service.printout_delay
+
+function Base.show(io::IO, qpu::AnyonQPU)
+    println(io, "Quantum Processing Unit:")
+    println(io, "   manufacturer:  $(qpu.manufacturer)")
+    println(io, "   generation:    $(qpu.generation) ")
+    println(io, "   serial_number: $(qpu.serial_number) ")
+end
+
+"""
+    VirtualQPU
+
+A data structure to represent a Quantum Simulator.  
+# Fields
+- `developers   ::String`   -- Simulator developers.
+- `package      ::String`   -- name of the underlying library package.
+
+# Example
+```jldoctest
+julia> qpu=VirtualQPU("Anyon Systems Inc.","Snowflake.jl")
+Quantum Simulator:
+   developers:  Anyon Systems Inc.
+   package:     Snowflake.jl
+
+
+```
+"""
+struct VirtualQPU <: AbstractQPU
+    developers  ::String
+    package     ::String
+end
+
+function Base.show(io::IO, qpu::VirtualQPU)
+    println(io, "Quantum Simulator:")
+    println(io, "   developers:  $(qpu.developers)")
+    println(io, "   package:     $(qpu.package)")
+end
+
+function read_response_body(body::Vector{UInt8})::String
+    # convert response body from binary to ASCII
+    read_buffer=IOBuffer(reinterpret(UInt8, body))
+    body_string=String(readuntil(read_buffer, 0x00))
+
+    if length(body_string)!=length(body)
+        throw(ArgumentError("Server returned an erroneous message, with nul terminator before end of string."))
+    end
+
+    return body_string
+end
+
+"""
+    run_job(qpu::AnyonQPU, circuit::QuantumCircuit,num_repetitions::Integer)
+
+Run a circuit computation on a `QPU` service, repeatedly for the specified 
+number of repetitions (num_repetitions). Returns the histogram of the 
+completed circuit calculations, or an error message.
+
+# Example
+```jldoctest
+julia> c = Client(host="http://example.anyonsys.com",user="test_user",access_token="not_a_real_access_token",requestor=MockRequestor());
+  
+julia> qpu=AnyonQPU(client=c,manufacturer="Anyon Systems Inc.",generation="Yukon",serial_number="ANYK202201");
+
+julia> run_job(qpu,QuantumCircuit(qubit_count=3,gates=[sigma_x(3),control_z(2,1)]) ,100)
+Status: succeeded
+
+Dict{String, Int64} with 1 entry:
+  "001" => 100
+
+```
+"""
+function run_job(qpu::AnyonQPU, circuit::QuantumCircuit,num_repetitions::Integer)::Dict{String,Int}
+    
+    client=get_client(qpu)
+
+    circuitID=submit_circuit(client,circuit,num_repetitions)
+
+    status=get_status(client,circuitID;)
+
+    ref_time_print=Base.time_ns()
+    printout_delay=get_printout_delay(qpu)/1e6 #convert ms to ns
+ 
+    ref_time_query=Base.time_ns()
+    query_delay=100/1e6 #100 ms between queries to host
+
+    while true        
+        current_time=Base.time_ns()
+
+        if (current_time-ref_time_print)>printout_delay
+            println(status)
```

Comment:
Here is still a printout. Please only mark threads as resolved once a commit has been pushed.